Zend/zend_constants.h

```diff
@@ -32,10 +32,14 @@
 #define CONST_IS_RECURSIVE(c) (Z_CONSTANT_FLAGS((c)->value) & CONST_RECURSIVE)
 #define CONST_PROTECT_RECURSION(c) \
 	do { \
-		Z_CONSTANT_FLAGS((c)->value) |= CONST_RECURSIVE; \
+		/* Ignored for internal constants, GH-18463 */ \
+		if (ZEND_CONSTANT_MODULE_NUMBER(c) == PHP_USER_CONSTANT) { \
+			Z_CONSTANT_FLAGS((c)->value) |= CONST_RECURSIVE; \
+		} \
 	} while (0)
 #define CONST_UNPROTECT_RECURSION(c) \
 	do { \
+		/* No need to check the module, always safe to remove. */ \
 		Z_CONSTANT_FLAGS((c)->value) &= ~CONST_RECURSIVE; \
```

Comment:
I think it's better not to do that. If we didn't separate with ZTS, it's still not great to write to the field from multiple threads.